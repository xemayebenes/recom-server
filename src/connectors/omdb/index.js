import fetch from 'node-fetch';

const API_KEY = 'a3d09618';

const DATA_REQUEST_URL = 'http://www.omdbapi.com/';


const generateMovieUriRequest = id =>
    `${DATA_REQUEST_URL}?i=${id}&apikey=${API_KEY}`;

const generateSerieUriRequest = title =>
    `${DATA_REQUEST_URL}?t=${title}&apikey=${API_KEY}`;

export const fetchOMDBMovie = id =>
    fetch(generateMovieUriRequest(id))
    .then(response => response.json());

export const fetchOMDBSerie = title =>
    fetch(generateSerieUriRequest(title))
    .then(response => response.json())