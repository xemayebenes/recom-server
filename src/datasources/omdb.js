import { RESTDataSource } from 'apollo-datasource-rest';

const DATA_REQUEST_URL = 'http://www.omdbapi.com/';

export default class OmdbAPI extends RESTDataSource {
  constructor() {
    super();
    this.baseURL = DATA_REQUEST_URL;
  }

  willSendRequest(request) {
    request.params.set('apikey', 'a3d09618');
  }

  getMovie = async id => this.get('', { i: id });

  getSerie = async title => this.get('', { t: title });
}
