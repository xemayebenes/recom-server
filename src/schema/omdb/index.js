import { gql } from 'apollo-server-express';

export const Rating = gql`
  type Rating @cacheControl(maxAge: 5) {
    Source: String
    Value: String
  }
`;
export const FilmOMDBData = gql`
  type FilmOMDBData {
    Title: String
    Year: String
    Rated: String
    Released: String
    Runtime: String
    Genre: String
    Director: String
    Writer: String
    Actors: String
    Plot: String
    Language: String
    Country: String
    Awards: String
    Poster: String
    Ratings: [Rating]
    Metascore: String
    imdbRating: Float
    imdbVotes: Int
    imdbID: String
    Type: String
    DVD: String
    BoxOffice: String
    Production: String
    Website: String
    Response: Boolean
  }
`;

export const SerieOMDBData = gql`
  type SerieOMDBData @cacheControl(maxAge: 5) {
    Title: String
    Year: String
    Rated: String
    Released: String
    Runtime: String
    Genre: String
    Director: String
    Writer: String
    Actors: String
    Plot: String
    Language: String
    Country: String
    Awards: String
    Poster: String
    Ratings: [Rating]
    Metascore: String
    imdbRating: Float
    imdbVotes: Int
    imdbID: String
    Type: String
    totalSeasons: Int
    Response: Boolean
  }
`;
