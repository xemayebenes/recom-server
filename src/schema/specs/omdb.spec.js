import { Rating, FilmOMDBData, SerieOMDBData } from '../omdb';

describe('SCHEMA OMDB', () => {
    describe('Rating', () => {
        it('should match snapshot', () => {
            expect(Rating).toMatchSnapshot();
        });
    });
    describe('FilmOMDBData', () => {
        it('should match snapshot', () => {
            expect(FilmOMDBData).toMatchSnapshot();
        });
    });
    describe('SerieOMDBData', () => {
        it('should match snapshot', () => {
            expect(SerieOMDBData).toMatchSnapshot();
        });
    });

});