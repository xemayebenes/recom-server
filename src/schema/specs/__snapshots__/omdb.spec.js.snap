// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCHEMA OMDB FilmOMDBData should match snapshot 1`] = `
"
    type FilmOMDBData {
        Title: String
        Year: String
        Rated: String
        Released: String
        Runtime: String
        Genre: String
        Director: String
        Writer: String
        Actors: String
        Plot: String
        Language: String
        Country: String
        Awards: String
        Poster: String
        Ratings: [Rating]
        Metascore: String
        imdbRating: Float
        imdbVotes: Int
        imdbID: String
        Type: String
        DVD: String
        BoxOffice: String
        Production: String
        Website: String
        Response: Boolean
    }
"
`;

exports[`SCHEMA OMDB Rating should match snapshot 1`] = `
"
    type Rating @cacheControl(maxAge: 5){
        Source: String
        Value:String
    }
"
`;

exports[`SCHEMA OMDB SerieOMDBData should match snapshot 1`] = `
"
    type SerieOMDBData @cacheControl(maxAge: 5){  
        Title: String
        Year: String
        Rated: String
        Released: String
        Runtime: String
        Genre: String
        Director: String
        Writer: String
        Actors: String
        Plot: String
        Language: String
        Country: String
        Awards: String
        Poster: String
        Ratings: [Rating]
        Metascore: String
        imdbRating: Float
        imdbVotes: Int
        imdbID: String
        Type: String
        totalSeasons: Int
        Response: Boolean
    }
"
`;
